lua/diffview/ui/panel.lua

```diff
@@ -288,7 +275,7 @@ function Panel:open()
   end
 
   self:resize()
-  utils.set_local(self.winid, self:class().winopts)
+  utils.set_local(self.winid, config.winopts)
```

Comment:
Thanks for the PR!

I think we can simplify this a lot by leaving the window options defined in the various panel classes, and treat `win_config.winopts` as overrides. Just apply the `win_config` window options after the classes own options:


```suggestion
  utils.set_local(self.winid, self:class().winopts)
  utils.set_local(self.winid, config.winopts)
```

---

Could you also please [amend the config validation](https://github.com/sindrets/diffview.nvim/blob/e32de2be00e0b5e23736f906f3d6d9dca14fbeb0/lua/diffview/ui/panel.lua#L160-L202) to ensure that `winopts` is a table (`:h vim.validate()`)?

I know I was the one who suggested the name `winopts`, but let's rename it to `win_opts`. Otherwise it would be strangely inconsistent with the naming of all the other config options.

Reply:
Great shout! Can't believe how much simpler that's made it :)